### PR TITLE
Add new events/adjust, remove binary sensors

### PR DESCRIPTION
**Pull request by kingsleyadam**
This makes a number of adjustments to events and binary sensors.

- Removes the `SwitchSensor` and `ForceOnOffSensor` from the binary_sensor as these do not make sense as sensors. They're moment in time "events" and will be treated just as events.
- Add the following new events
  - `BlindSensor`
  - `DimmingSensor`
- Also, adjust the sensors for `SwitchSensor`, `ForceOnOffSensor` to use the new state options.

### CODE REVIEW

File: custom_components/abbfreeathome_ci/translations/en.json [hunk 1]
```diff
@@ -88,12 +96,30 @@
...
+        "name": "Dimming Event ({channel_id})",
+        "state_attributes": {
+          "event_type": {
+            "state": {
+              "longpress_down": "Dimming Down",
+              "longpress_down_release": "Dimming Down Release",
+              "longpress_up": "Dimming Up",
+              "longpress_up_release": "Dimming Up Release",
+              "off": "Off",
+              "on": "On",
```
Comment (derjoerg) [inline custom_components/abbfreeathome_ci/translations/en.json:109]: Are the `on` and `off` translations needed? As they are already referenced to the common translations

**Comment by derjoerg** [comment]
closes #73 #77

**Comment by kingsleyadam** [comment]
> closes #73 #77

I'm not sure which functions you have, but if you could test this branch against your environment that'd be great.

File: custom_components/abbfreeathome_ci/translations/en.json (hunk 1 above)
Comment (derjoerg) [reply custom_components/abbfreeathome_ci/translations/en.json:109]: > Yea, they are. Without this then the On/Off event's won't be translated.

But then I don't get why referencing in strings.json the common-translations `"off": "[%key:common::state::off%]"`. My understanding was, when referncing the common-translations we don't need to care about them and don't need to do double work. Or have I completely misunderstood that?

Comment (kingsleyadam) [reply custom_components/abbfreeathome_ci/translations/en.json:109]: With custom integrations the `strings.json` file isn't used at all, just the language specific files. If this is submitted as a core integration the opposite will be true, you'd only provide the `strings.json` file and no specific languages files.

When I run the integration in my dev environment it automatically generates the language files, which is what I place into this repo to be committed and used.

To be ready to be submitted as a core integration we need to maintain both in this repo.

**Comment by derjoerg** [comment]
> 😢 I have to admit that I have no glue how to do this. I have my fork linked in my dev-environment. Do you know how I can (in visual studo code) fetch a branch of your repo?

Doesn't matter, I've applied the changes manually :smiley:

**Comment by kingsleyadam** [comment]
> I have to admit that I have no glue how to do this. I have my fork linked in my dev-environment.
> Do you know how I can (in visual studo code) fetch a branch of your repo?

Here's a short write-up from GitHub that's pretty good: https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories

You'd need to add my repo as a remote:

```
git remote add kingsleyadam https://github.com/kingsleyadam/local-abbfreeathome-hass.git
```

Then you can pull/fetch from that new remote

```
git fetch kingsleyadam
```

Then you can checkout this branch,

```
git switch -c u/kingsleyadam/adjustEvents kingsleyadam/u/kingsleyadam/adjustEvents
```

Hopefully that works. I didn't test each command. :)

**Comment by derjoerg** [comment]
ForceOnOffSensor is working
SwitchSensor is working
BlindSensor is working
DimmingSensor is not working

First of all: My DimmingSensors have two events: Dimming Event and Switch Event
Up and Down is working (well, both sensors are updating with the correct On/Off event)

But as soon as I try the longpress, I get the following error
```
Traceback (most recent call last):
  File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/abbfreeathome/freeathome.py", line 174, in ws_listen
    await self.api.ws_listen(callback=self.update_device)
  File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/abbfreeathome/api.py", line 261, in ws_listen
    await self.ws_receive(callback, retry_interval)
  File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/abbfreeathome/api.py", line 287, in ws_receive
    await callback(_ws_data)
  File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/abbfreeathome/freeathome.py", line 182, in update_device
    _device.update_device(_datapoint_key, _datapoint_value)
  File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/abbfreeathome/devices/base.py", line 115, in update_device
    callback()
  File "/workspaces/homeassistant_core/config/custom_components/abbfreeathome_ci/event.py", line 141, in _async_handle_event
    self._trigger_event(event_type)
  File "/workspaces/homeassistant_core/homeassistant/components/event/__init__.py", line 151, in _trigger_event
    raise ValueError(f"Invalid event type {event_type} for {self.entity_id}")
ValueError: Invalid event type longpress_down for event.sensoreinheit_1_fach_licht_switch_event_ch0000_3
```
Which is logical, as the switch_event don't know this event-type.

But I don't understand how this can happen (the two events) when looking at the code

**Comment by derjoerg** [comment]
> But I don't understand how this can happen (the two events) when looking at the code

The problem seems to be in the pypi library.

Loading all SwitchSensors will also include all DimmingSensors.
Loading all DimmingSensors will only have DimmingSensors.

**Comment by kingsleyadam** [comment]
> ForceOnOffSensor is working
> 
> SwitchSensor is working
> 
> BlindSensor is working
> 
> DimmingSensor is not working
> 
> 
> 
> First of all: My DimmingSensors have two events: Dimming Event and Switch Event
> 
> Up and Down is working (well, both sensors are updating with the correct On/Off event)
> 
> 
> 
> But as soon as I try the longpress, I get the following error
> 
> ```
> 
> Traceback (most recent call last):
> 
>   File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/abbfreeathome/freeathome.py", line 174, in ws_listen
> 
>     await self.api.ws_listen(callback=self.update_device)
> 
>   File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/abbfreeathome/api.py", line 261, in ws_listen
> 
>     await self.ws_receive(callback, retry_interval)
> 
>   File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/abbfreeathome/api.py", line 287, in ws_receive
> 
>     await callback(_ws_data)
> 
>   File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/abbfreeathome/freeathome.py", line 182, in update_device
> 
>     _device.update_device(_datapoint_key, _datapoint_value)
> 
>   File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/abbfreeathome/devices/base.py", line 115, in update_device
> 
>     callback()
> 
>   File "/workspaces/homeassistant_core/config/custom_components/abbfreeathome_ci/event.py", line 141, in _async_handle_event
> 
>     self._trigger_event(event_type)
> 
>   File "/workspaces/homeassistant_core/homeassistant/components/event/__init__.py", line 151, in _trigger_event
> 
>     raise ValueError(f"Invalid event type {event_type} for {self.entity_id}")
> 
> ValueError: Invalid event type longpress_down for event.sensoreinheit_1_fach_licht_switch_event_ch0000_3
> 
> ```
> 
> Which is logical, as the switch_event don't know this event-type.
> 
> 
> 
> But I don't understand how this can happen (the two events) when looking at the code

Can you share webs device config for these?

I find it strange they have both events. I'd think they'd have one or the other.

**Comment by derjoerg** [comment]
> I find it strange they have both events. I'd think they'd have one or the other.

I think I found the problem

look at https://github.com/kingsleyadam/local-abbfreeathome/issues/112

**Comment by kingsleyadam** [comment]
> I think I found the problem
> 
> look at [kingsleyadam/local-abbfreeathome#112](https://github.com/kingsleyadam/local-abbfreeathome/issues/112)

I've pushed a patch for that issue and bumped the PyPi package version, can you give this one a try?

**Comment by derjoerg** [comment]
> 
> I've pushed a patch for that issue and bumped the PyPi package version, can you give this one a try?

Now all sensors (events) are working as expected. Very, very nice

**Pull request merged**